### ISSUE TITLE
Differential pair router only likes horizontal and vertical vias

### ISSUE THREAD

**Issue by rroohhh** [horizon-eda/horizon#434]
The p&s router puts the vias for differential pairs either on the same x or same y coordinate. This leads to strange via placement for parts at an angle.

![2020-06-22-202031_3200x1080_scrot](https://user-images.githubusercontent.com/7414518/85322147-1c53ce00-b4c6-11ea-996a-48eeadb788b4.png)

I think it should atleast allow for 45 degree angles aswell.

**Comment by carrotIndustries** [comment]
> I think it should atleast allow for 45 degree angles aswell.

It does in some places. As the router is more or less a black box to me, I don't think there's much I can do about it...

**Comment by rroohhh** [comment]
Right, I wasn't sure, if this was a thing the router does or something horizon does.

I guess I can try digging around a bit in the router.

**Comment by bob-u** [comment]
Maybe changing grid orientation, as @carrotIndustries mentioned might be possible, could change router's via orientation also.

**Comment by phkahler** [comment]
I wonder if that's trying to maintain equal lengths as well. obviously putting them on a 45 would allow perfect symmetry on this side of the board, but maybe it's part of a larger solution involving another layer.
